includes/class-analytics.php

```diff
@@ -47,49 +47,85 @@ public function __construct() {
 	}
 
 	/**
-	 * Tell Site Kit to report the article's category as custom dimension.
-	 * More about custom dimensions:
-	 * https://support.google.com/analytics/answer/2709828.
+	 * Tell Site Kit to report the article's data as custom dimensions.
+	 * More about custom dimensions: https://support.google.com/analytics/answer/2709828.
 	 */
-	public static function handle_category_reporting() {
-		$category_dimension_id = get_option( Analytics_Wizard::$category_dimension_option_name );
-		if ( ! $category_dimension_id ) {
-			return;
-		}
-		// Remove `ga:` prefix.
-		$category_dimension_id = substr( $category_dimension_id, 3 );
-
-		$categories = get_the_category();
-		if ( ! empty( $categories ) ) {
-			$categories_slugs = implode(
-				',',
-				array_map(
-					function( $cat ) {
-						return $cat->slug;
-					},
-					$categories
-				)
-			);
-			// Non-AMP.
-			add_filter(
-				'googlesitekit_gtag_opt',
-				function ( $gtag_opt ) use ( $categories_slugs, $category_dimension_id ) {
-					$gtag_opt[ $category_dimension_id ] = $categories_slugs;
-					return $gtag_opt;
+	public static function handle_custom_dimensions_reporting() {
+		$custom_dimensions = Analytics_Wizard::list_configured_custom_dimensions();
+		foreach ( $custom_dimensions as $dimension ) {
+			$dimension_role = $dimension['role'];
+			// Remove `ga:` prefix.
+			$dimension_id = substr( $dimension['id'], 3 );
+
+			$post = get_post();
+			if ( $post ) {
+				if ( 'category' === $dimension_role ) {
+					$categories = get_the_category();
+					if ( ! empty( $categories ) ) {
+						$categories_slugs = implode(
+							',',
+							array_map(
+								function( $cat ) {
+									return $cat->slug;
+								},
+								$categories
+							)
+						);
+						self::add_custom_dimension_to_ga_config( $dimension_id, $categories_slugs );
+					}
 				}
-			);
-			// AMP.
-			add_filter(
-				'googlesitekit_amp_gtag_opt',
-				function ( $gtag_amp_opt ) use ( $categories_slugs, $category_dimension_id ) {
-					$tracking_id = $gtag_amp_opt['vars']['gtag_id'];
-					$gtag_amp_opt['vars']['config'][ $tracking_id ][ $category_dimension_id ] = $categories_slugs;
-					return $gtag_amp_opt;
+
+				if ( 'author' === $dimension_role ) {
+					$author_id = $post->post_author;
+					self::add_custom_dimension_to_ga_config(
+						$dimension_id,
+						get_the_author_meta( 'first_name', $author_id ) . ' ' . get_the_author_meta( 'last_name', $author_id )
+					);
 				}
-			);
+
+				if ( 'word_count' === $dimension_role ) {
+					self::add_custom_dimension_to_ga_config(
+						$dimension_id,
+						count( explode( ' ', wp_strip_all_tags( $post->post_content ) ) )
+					);
+				}
+
+				if ( 'publish_date' === $dimension_role ) {
+					self::add_custom_dimension_to_ga_config(
+						$dimension_id,
+						date_format( date_create( $post->post_date ), 'Y-m-d' )
```

Comment:
Just a nitpick and not at all a blocker, but couldn't we just do `get_the_time( 'Y-m-d', $post->ID )`?

Reply:
Good idea, changed in 34b956172e6b0c8b70c06eec93ff46ad987c57a0. Also added time, as suggested in #660